src/services/MenuBuilder.ts

```diff
@@ -55,14 +54,18 @@ export class MenuBuilder {
     const items: ContentItemModel[] = [];
     const tagsMap = MenuBuilder.getTagsWithOperations(spec);
 
-    let setionToTheEnd = '';
+    let sectionsBefore = '';
+    let sectionsAfter = '';
 
-    options.sectionsAtTheEnd.forEach(function(s) {
-      setionToTheEnd += extractContent(spec.info.description || '', s);
-      spec.info.description = removeContent(spec.info.description || '', s);
+    new MarkdownRenderer(options).extractHeadings(spec.info.description || '').forEach(function(h) {
```

Comment:
Looks like you misunderstood me. You can get all markdown items using 

```js
const mdHeadings = MenuBuilder.addMarkdownItems(spec.info.description || '', undefined, 1, options)
```

Then just split it into two arrays by item `.name` and push **before** part here and **after** part after.

Reply:
Oh, sorry! Yes, I misunderstood you. I will change it

Reply:
Thanks!

Reply:
Something like this?

```javascript
    const mdHeadings = MenuBuilder.addMarkdownItems(spec.info.description || '', undefined, 1, options);
    const mdHeadingsBefore = mdHeadings.filter(h => !options.sectionsAtTheEnd.includes(h.name));
    const mdHeadingsAfter = mdHeadings.filter(h => options.sectionsAtTheEnd.includes(h.name));

    items.push(...mdHeadingsBefore);

    [...]

    items.push(...mdHeadingsAfter);
```

Reply:
Yes!